doc/devices/drivers/gpio/gpio.rst

```diff
@@ -0,0 +1,294 @@
+    GPIO pin polarity is normal.
+
+    PINCTRL pin configuration property:
+
+    - :c:macro:`PINCTRL_CONFIG_OUTPUT_HIGH`
+
+**GPIO_POL_INV**
+    GPIO pin polarity is inverted.
+
+    PINCTRL pin configuration property:
+
+    - :c:macro:`PINCTRL_CONFIG_OUTPUT_LOW`
+
+**GPIO_PUD_NORMAL**
+    GPIO pin to have no pull-up or pull-down.
+
+    PINCTRL pin configuration property:
+
+    - :c:macro:`PINCTRL_CONFIG_BIAS_DISABLE`
+
+**GPIO_PUD_PULL_UP**
+    Enable GPIO pin pull-up.
+
+    PINCTRL pin configuration property:
+
+    - :c:macro:`PINCTRL_CONFIG_BIAS_PULL_UP`
+
+**GPIO_PUD_PULL_DOWN**
+    Enable GPIO pin pull-down.
+
+    PINCTRL pin configuration property:
+
+    - :c:macro:`PINCTRL_CONFIG_BIAS_PULL_DOWN`
+
+**GPIO_DS_DFLT_LOW**
+    Default drive strength standard when GPIO pin output is low.
+
+    PINCTRL pin configuration property:
+
+    - :c:macro:`PINCTRL_CONFIG_DRIVE_STRENGTH_DEFAULT`
+
+**GPIO_DS_DFLT_HIGH**
+    Default drive strength when GPIO pin output is high.
+
+    PINCTRL pin configuration property:
+
+    - :c:macro:`PINCTRL_CONFIG_DRIVE_STRENGTH_DEFAULT`
+
+**GPIO_DS_ALT_LOW**
+    Alternative drive strength when GPIO pin output is low.
+
+    PINCTRL pin configuration property:
+
+    - :c:macro:`PINCTRL_CONFIG_DRIVE_STRENGTH_7`
+
+**GPIO_DS_ALT_HIGH**
+    Alternative drive strength when GPIO pin output is high.
+
+    PINCTRL pin configuration property:
+
+    - :c:macro:`PINCTRL_CONFIG_DRIVE_STRENGTH_7`
+
+**GPIO_DS_DISCONNECT_LOW**
+    Disconnect pin when GPIO pin output is low.
+
+    PINCTRL pin configuration property:
+
+    - :c:macro:`PINCTRL_CONFIG_DRIVE_OPEN_SOURCE`
+
+**GPIO_DS_DISCONNECT_HIGH**
+    Disconnect pin when GPIO pin output is high.
+
+    PINCTRL pin configuration property:
+
+    - :c:macro:`PINCTRL_CONFIG_DRIVE_OPEN_DRAIN`
+
+Signalling Configuration
+========================
+
+Pin signalling can be configured by the general GPIO configuration interface.
+
+:c:func:`gpio_pin_configure`
+    Set the configuration of a pin.
+
+A generic set of GPIO port pin signalling properties is defined in
+:file:`include/gpio.h`. A GPIO driver may only support a subset of the
+signalling properties.
+
+**GPIO_INT**
+    GPIO pin to trigger interrupt.
+
+**GPIO_INT_ACTIVE_LOW**
+    GPIO pin trigger on level low or falling edge.
+
+**GPIO_INT_ACTIVE_HIGH**
+    GPIO pin trigger on level high or rising edge.
+
+**GPIO_INT_CLOCK_SYNC**
+    GPIO pin trigger to be synchronized to clock pulses.
+
+**GPIO_INT_DEBOUNCE**
+    Enable GPIO pin trigger debounce.
+
+**GPIO_INT_LEVEL**
+    Do Level trigger.
+
+**GPIO_INT_EDGE**
+    Do Edge trigger.
+
+**GPIO_INT_DOUBLE_EDGE**
+    Interrupt triggers on both rising and falling edge.
+
+
+GPIO device tree support
+************************
+
+Nodes
+=====
+
+GPIO Controller Node
+--------------------
+
+The GPIO device is defined by a GPIO controller node.
+
+Required properties are:
+
+    - gpio-controller: Indicates this device is a GPIO controller
```

Comment:
outdent these three list items so the `-` lines up with the ``R`` in ``Required``.